persistence/sql/persister_jwk.go

```diff
@@ -165,9 +167,19 @@ func (p *Persister) GetKeySet(ctx context.Context, set string) (*jose.JSONWebKey
 }
 
 func (p *Persister) DeleteKey(ctx context.Context, set, kid string) error {
-	return sqlcon.HandleError(p.Connection(ctx).RawQuery("DELETE FROM hydra_jwk WHERE sid=? AND kid=?", set, kid).Exec())
+	count, err := p.Connection(ctx).RawQuery("DELETE FROM hydra_jwk WHERE sid=? AND kid=? AND nid = ?", set, kid, p.NetworkID(ctx)).ExecWithCount()
+	if count == 0 {
```

Comment:
These checks are new. This is considered a regression. DELETE in REST usually always returns 204 No Content because the deletion succeeded if the object does not exist (because nothing was there to delete in the first place). I would suggest to revert these checks with count == 0 whereever they were introduced UNLESS there is a very good reason to do this, and it is documented as a brekaing change. If it is, please link the location of that documentation in this comment.

Reply:
There are two reasons I introduced this change - to facilitate a test, and because in my experience, it is the more common practice. I thought the current implementation was a mistake, but I appreciate being informed that the current state is intentional. I'll revert my change to preserve the convention.

Reply:
Thank you for the thoughtful response!